c/ovm.c

```diff
@@ -890,18 +874,18 @@ static word prim_sys(int op, word a, word b, word c) {
 
 static word prim_lraw(word wptr, int type, word revp) {
```

Comment:
revp is now unnecessary and could be removed from the primop and here

Reply:
Can you please have a look at my tree (https://github.com/jo-he/owl-lisp/commit/434cfaf0a2c00dda9cb26e554c85259ab12db1b1)? I tried to do it the same way as 'cons' (because they use the same input/output). But I couldn't work it out so far. Actually, I don't know, if this is correct, because I don't understand the list behind the 60 in (func '(3 60 4 5 6 24 6)).
I guess we need a new init.fasl, because of the arity errors. But trying to build one, the best results I got was exit code 1 or 127. You probably have a good hint for me (-;

Reply:
Ah, good. The bytecode is roughtly: arity 3, lraw(r4, r5) -> r6, ret(r6). When you start the compiler after modifying one of the core instructions, the compiler image will contain the old version of raw having more arguments. This leads to instruction decoding getting off sync and surely interesting times.

A good way to avoid this in a language the compiles itself is to get to a point, where both the old and new functionality are present. Then you can switch all places to use the new code and drop the old one.

Here you could proceed as follows: rename raw everywhere to old-raw, add opcode 37 to vm with updated raw, define the new raw with bytecode '(2 37 4 5 24 5), switch all uses of old-raw to the new raw while removing the extra argument, and finally remove the definition of old-raw.

At this point the code should be merged, as it can still be used to build owl from current init.fasl, since the old opcode is present in the vm, but the new fasl images will no longer use it. Then I could update the init fasl and you could remove the old instruction from the vm.

Fasl updates allow arbitrary code to be injected to the heap. For security reasons there should therefore be a commit before each fasl update, which links the new heap to the previous one.